### PR TITLE
Crucial updates for Blockchain templates

### PR DESCRIPTION
**Pull request by Duality-CDOO**
Anyone attempting to deploy using the blockchain templates currently will not be able to set up a Bitcoin node, Dash node, connect to the correct Syscoin blockchain, and Shadowcoin has been replaced by Particl.

These templates must be updated ASAP.

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Update Syscoin to version 4.0.3 and connect to the correct blockchain
* Add quickstart button to syscoin.md and add syscoin.png
* Updated all API's used in azuredeploy.json to latest versions
* Updated blockchain/metadata.json githubUsername and dateUpdated
* Refactored azuredeploy.json and removed Shadowcoin as it has changed/rebranded to Particl
* Add Bitcoin to template (md, image, quickstart, install script, azuredeploy.json)
* Add Dash to template (md, image, quickstart, install script, azuredeploy.json)
* Add Dynamic to template (md, image, quickstart, install script, azuredeploy.json)
* Add Sequence to template (md, image, quickstart, install script, azuredeploy.json)
* Add Particl to template (md, image, quickstart, install script, azuredeploy.json)

### CODE REVIEW

**Comment by Duality-CDOO** [comment]
@bmoore-msft

Here is the resubmitted PR as requested.

**Comment by Duality-CDOO** [comment]
This is failing the 'Run Best Practice Tests' and 'Deploy mainTemplate' tests.

The error is that _artifactsLocationSasToken is not referenced.

image

The azuredeploy.json file does have this declared:
```
"_artifactsLocation": {
      "type": "string",
      "metadata": {
        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
      },
      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/blockchain/"
    },
    "_artifactsLocationSasToken": {
      "type": "securestring",
      "metadata": {
        "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
      },
      "defaultValue": ""
```
image

I really require help getting this to pass the tests if anyone knows how to fix this issue.

This isn't something I have introduced but was already present.

File: blockchain/azuredeploy.json [hunk 1]
```diff
@@ -100,7 +103,7 @@
     "addressPrefix": "10.0.0.0/16",
     "imagePublisher": "Canonical",
     "imageVersion": "latest",
-    "imageSKU": "14.04.4-LTS",
+    "imageSKU": "18.04.3-LTS",
```
Comment (bmoore-msft) [inline blockchain/azuredeploy.json]: this image doesn't exist - you can use 18.04-LTS

Comment (Duality-CDOO) [reply blockchain/azuredeploy.json]: Fixed.

**Comment by Duality-CDOO** [comment]
> see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts
> 
> to fix the BP error on line 259...

I am still confused about what is needed with this change.

**Comment by bmoore-msft** [comment]
looks like you got past the first error - the second one isn't related to your change but if the change the broke that is reverted your change still won't work...  because you're adding support for new software types.

I can't make changes to your branch (you need to allow maintainers changes perms)...  I might be able to fix in master, but then you'll have to resolve merge conflicts.

**Comment by Duality-CDOO** [comment]
> looks like you got past the first error - the second one isn't related to your change but if the change the broke that is reverted your change still won't work... because you're adding support for new software types.
> 
> I can't make changes to your branch (you need to allow maintainers changes perms)... I might be able to fix in master, but then you'll have to resolve merge conflicts.

If you could PR to my master branch, then there should be no conflicts.

**Comment by bmoore-msft** [comment]
I submitted a fix for that deployment failure to Azure/master... so you can pull from there or just resolve the merge changes from here.

**Comment by Duality-CDOO** [comment]
> I submitted a fix for that deployment failure to Azure/master... so you can pull from there or just resolve the merge changes from here.

Hoping for a pass this time around @bmoore-msft

**Comment by Duality-CDOO** [comment]
```2020-02-28T00:00:52.8292660Z ##[error]D:\a\1\test\ci-scripts\Validate-Metadata.ps1 : Cannot process argument transformation on parameter 'BuildReason'. Cannot convert value "System.String" to type "System.Management.Automation.SwitchParameter". Boolean parameters accept only Boolean values and numbers, such as $True, $False, 1 or 0.
```

**Comment by Duality-CDOO** [comment]
Your test file test\ci-scripts\Validate-Metadata.ps1 is causing an issue @bmoore-msft

**Comment by Duality-CDOO** [comment]
```W: GPG error: http://archive.ubuntu.com/ubuntu bionic InRelease: Splitting up /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_bionic_InRelease into data and signature failed
E: The repository 'http://archive.ubuntu.com/ubuntu bionic InRelease' is not signed.```

**Comment by Duality-CDOO** [comment]
Not sure if the tests have timed out or if it is another error @bmoore-msft 

```
2:39:25 AM - Provisioning of VM extension newuserscript has timed out. Extension provisioning has taken too long to complete. The extension last reported "Plugin enabled".

More information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot
2:39:25 AM - Template output evaluation skipped: at least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.
2:39:25 AM - Template output evaluation skipped: at least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.
D:\a\1\test\Deploy-AzTemplate.ps1 : Deployment failed.
At D:\a\_temp\c2d90632-7952-4603-a528-08800455cc63.ps1:2 char:1
+ d:\a\1\test\Deploy-AzTemplate.ps1 `
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
+ FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Deploy-AzTemplate.ps1
```

**Comment by Duality-CDOO** [comment]
@bmoore-msft This now passes all the tests. Thank you for all your help.

**Comment by Duality-CDOO** [comment]
@bmoore-msft This now passes all the tests. Thank you for all your help.

**Pull request merged**